src/github.com/couchbaselabs/sync_gateway/base/dcp_feed.go

```diff
@@ -0,0 +1,193 @@
+//  Copyright (c) 2014 Couchbase, Inc.
+//  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+//  except in compliance with the License. You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+//  Unless required by applicable law or agreed to in writing, software distributed under the
+//  License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+//  either express or implied. See the License for the specific language governing permissions
+//  and limitations under the License.
+
+package base
+
+import (
+	"encoding/json"
+	"sync"
+
+	"github.com/couchbase/gomemcached"
+	"github.com/couchbaselabs/go-couchbase/cbdatasource"
+	"github.com/couchbaselabs/walrus"
+)
+
+type couchbaseDCPFeedImpl struct {
+	bds    cbdatasource.BucketDataSource
+	events <-chan walrus.TapEvent
+}
+
+func (feed *couchbaseDCPFeedImpl) Events() <-chan walrus.TapEvent {
+	return feed.events
+}
+
+func (feed *couchbaseDCPFeedImpl) Close() error {
+	return feed.bds.Close()
+}
+
+// Implementation of couchbase.AuthHandler for use by cbdatasource during bucket connect.
+type dcpAuth struct {
+	Username, Password, BucketName string
+}
+
+// GetCredentials is based on approach used in go-couchbase pools.go authHandler().
+func (a dcpAuth) GetCredentials() (string, string, string) {
+	if a.Username == "" {
+		return a.BucketName, "", a.BucketName
```

Comment:
Is there any docs that say GetCredentials() should return bucketname for username field if username is blank?  If not, maybe we should add some.

I didn't see any explanation in go-couchbase:

```
// AuthHandler is a callback that gets the auth username and password
// for the given bucket.
type AuthHandler interface {
	GetCredentials() (string, string, string)
}

```

how did you even know to use that "trick"?